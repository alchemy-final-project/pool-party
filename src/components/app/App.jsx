import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import OwnerSignUp from '../../containers/OwnerSignUp';
import TenantLogin from '../../containers/TenantLogin';
import TenantSignUp from '../../containers/TenantSignUp';
import Payment from '../../containers/Payment';
import Dashboard from '../../containers/Dashboard';
import Header from '../header/Header';

function App() {
  return (
    <>
      <Header/>
      <Router>
        <Switch>
          <Route exact path="/" component={TenantLogin} />
          <Route exact path="/tenantSignup" component={TenantSignUp} />
          <Route exact path="/ownerSignup" component={OwnerSignUp} />
          <Route exact path="/payment" component={Payment} />
          <Route exact path="/dashboard" component={Dashboard} />
        </Switch>
      </Router>
    </>
  );
}
export default App;
