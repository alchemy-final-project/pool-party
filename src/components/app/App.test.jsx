import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import {
  Elements
} from '@stripe/react-stripe-js';
import { loadStripe } from '@stripe/stripe-js';

const stripePromise = loadStripe('pk_test_51IQLH1AlZQgGODSJBkoELcpjSB9liGgfLpot9mWQER1Cv9RCBpLnvhdrDPORukS3YQoeQ4OHq04EC95pX1GrnAHY00rW9sayk4');

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(
      <Elements stripe={stripePromise}>
        <App />
      </Elements>);
    expect(asFragment()).toMatchSnapshot();
  });
});
