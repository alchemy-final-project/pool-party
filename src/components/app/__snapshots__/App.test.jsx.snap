// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div>
    <h1>
      POOL PARTY + Navigation
    </h1>
  </div>
  <div>
    <div>
      <img
        src="/path/image"
      />
    </div>
    <div>
      <form
        class="loginForm"
      >
        <label>
          Email
        </label>
        <input
          placeholder="Enter email as username"
          required=""
          type="plain/text"
          value=""
        />
        <label>
          Password
        </label>
        <input
          placeholder="Type Password"
          required=""
          type="password"
          value=""
        />
        <button>
          Log In
        </button>
      </form>
    </div>
    <a
      href="/tenantSignUp"
    >
      Need to signup? Click here.
    </a>
  </div>
</DocumentFragment>
`;
